### DISCUSSION TITLE
Trame vs Plotly Dash

### DISCUSSION THREAD

**Discussion by JoseRamonRodriguez** [Kitware/trame#130] [Q&A] [answered]
Hi all,

I'm quite new on Trame but I have been using Dash for a while to build some web applications. I really like how easy you can create interactive 3D visualization with Trame and I wondered whether I should start using it, instead of Dash. How would you compare generally both tools? When do you think it is better to use one among the other? Could they be complementary tools somehow?

Thanks in advance!

**Comment by jourdain** [comment, marked as answer]
The main and key difference (putting aside the underlying technology) is that __trame is stateful__ while __dash is stateless__.

That simple statement has many implications. But in a nutshell, __dash__ is meant to be deployed as a web site and scale to serve many client with the same server (or with std web scaling infrastructure) while __trame__ is more about creating capable and highly interactive applications which can then be used locally on desktop, jupyter tool or deployed as a service where each client get their own process. 

As you can imagine, deploying a trame application in the cloud is slightly more complex than doing the same for dash but we have several path on how to handle that. Also trame can be use for collaborative usage when you allow several clients to connect to the same server process.

Another way to see trame could be the framework that enable anyone to create desktop applications which can then be deployed as SaaS while keeping a single code base.

For full feature desktop usage we are currently working on streamlining the [integration](https://github.com/Kitware/trame-tauri) of [tauri](https://tauri.app/) on top of the currently supported [pywebview](https://pywebview.flowrl.com/) (arg: `--app`).

Then I can start going down into the underlying technology but at that point it might drift into a troll talk. ;-)

I hope that helps and clarify some of the core aspects.

Seb

**Reply by JoseRamonRodriguez** [reply]
Hi Sebastien,

Great, thank you very much for the detailed response! And really good job guys ;)

**Reply by jourdain** [reply]
Thanks Jose! I'm glad you see value in trame...

**Reply by jourdain** [reply]
If you feel like it please consider adding a comments to either [Used by](https://github.com/Kitware/trame/issues/19) or [testimonial](https://github.com/Kitware/trame/issues/18). It is probably too early for your usage but that would help trame get some recognition.

**Reply by JoseRamonRodriguez** [reply]
Thanks. We are going to evaluate it internally first, in order to understand if it will fit for our purpose. After that, sure we can add those comments if needed!
